### PR TITLE
feat: update to Angular 9

### PR DESCRIPTION
**Pull request by yggg**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/nebular/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:

### CODE REVIEW

**Comment by vishnu-dev** [comment]
Eagerly waiting for this. Please do it soon. Thank you so much for a wonderful library.

File: package.json [hunk 1]
```diff
@@ -85,14 +86,15 @@
...
   "dependencies": {
-    "@angular/animations": "8.0.0",
-    "@angular/common": "8.0.0",
-    "@angular/compiler": "8.0.0",
-    "@angular/core": "8.0.0",
-    "@angular/forms": "8.0.0",
-    "@angular/platform-browser": "8.0.0",
-    "@angular/platform-browser-dynamic": "8.0.0",
-    "@angular/router": "8.0.0",
+    "@angular/animations": "^9.0.2",
```
Comment (Tibing) [inline package.json]: It's already  :)
```suggestion
    "@angular/animations": "^9.0.3",
```

File: packages-smoke/angular.json [hunk 2]
```diff
@@ -12,31 +12,32 @@
...
         "@schematics/angular:service": {
-          "spec": false
+          "skipTests": true
         }
       },
       "architect": {
         "build": {
           "builder": "@angular-devkit/build-angular:browser",
           "options": {
+            "aot": true,
```
Comment (Tibing) [inline packages-smoke/angular.json]: Check is it required

File: packages-smoke/package.json [hunk 3]
```diff
@@ -15,16 +15,16 @@
...
     "@angular/cdk": "8.0.0",
-    "@angular/common": "8.0.0",
-    "@angular/compiler": "8.0.0",
-    "@angular/core": "8.0.0",
-    "@angular/forms": "8.0.0",
-    "@angular/platform-browser": "8.0.0",
-    "@angular/platform-browser-dynamic": "8.0.0",
-    "@angular/platform-server": "8.0.0",
-    "@angular/router": "8.0.0",
+    "@angular/common": "9.0.2",
```
Comment (Tibing) [inline packages-smoke/package.json]: 9.0.3

File: packages-smoke/src/main.server.ts [hunk 4]
```diff
@@ -10,3 +10,5 @@ export { AppServerModule } from './app/app.server.module';
 export { ngExpressEngine } from "@nguniversal/express-engine";
 export { provideModuleMap } from "@nguniversal/module-map-ngfactory-loader";
 
+
+export { renderModule, renderModuleFactory } from '@angular/platform-server';
```
Comment (Tibing) [inline packages-smoke/src/main.server.ts]: new line

File: scripts/gulp/tasks/build/compile/ts-configs.ts [hunk 5]
```diff
@@ -19,7 +19,8 @@ export function createTsConfigEsm2015(packageName): string {
     "skipMetadataEmit": false,
     "strictTypeChecks": true,
     "flatModuleOutFile": "index.js",
-    "flatModuleId": "@nebular/${packageName}"
+    "flatModuleId": "@nebular/${packageName}",
+    "enableIvy": false
```
Comment (Tibing) [inline scripts/gulp/tasks/build/compile/ts-configs.ts]: we need to compile libs to viewEngine and them user's project will run ngcc right?

File: src/framework/theme/components/cdk/table/row.ts [hunk 6]
```diff
@@ -116,7 +114,6 @@ export class NbFooterRowComponent extends CdkFooterRow {
     'role': 'row',
   },
   providers: [{ provide: CdkRow, useExisting: NbRowComponent }],
-  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (Tibing) [inline src/framework/theme/components/cdk/table/row.ts]: WHY?

File: src/framework/theme/components/cdk/table/row.ts [hunk 7]
```diff
@@ -100,7 +99,6 @@ export class NbHeaderRowComponent extends CdkHeaderRow {
     'class': 'nb-footer-row',
     'role': 'row',
   },
-  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (Tibing) [inline src/framework/theme/components/cdk/table/row.ts]: WHY?

File: src/framework/theme/components/cdk/table/row.ts [hunk 8]
```diff
@@ -85,7 +85,6 @@ export class NbRowDefDirective<T> extends CdkRowDef<T> {
     'class': 'nb-header-row',
     'role': 'row',
   },
-  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (Tibing) [inline src/framework/theme/components/cdk/table/row.ts]: WHY?

File: src/framework/theme/components/layout/layout.component.spec.ts [hunk 9]
```diff
@@ -66,7 +66,7 @@ describe('NbLayoutComponent', () => {
     // }));
     //
     // it('should add left padding to layout container in RTL mode when blocking scroll', fakeAsync(() => {
-    //   const layoutDirectionService: NbLayoutDirectionService = TestBed.get(NbLayoutDirectionService);
```
Comment (Tibing) [inline src/framework/theme/components/layout/layout.component.spec.ts]: Would you mind removing entire comment?) We have a version control system, we don't need to comment unused code.

File: src/framework/theme/components/tree-grid/tree-grid-row.component.ts [hunk 10]
```diff
@@ -19,7 +19,6 @@ export const NB_ROW_DOUBLE_CLICK_DELAY: number = 200;
     'role': 'row',
   },
   providers: [{ provide: NbCdkRow, useExisting: NbTreeGridRowComponent }],
-  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (Tibing) [inline src/framework/theme/components/tree-grid/tree-grid-row.component.ts]: WHYYYY&

Comment (yggg) [reply src/framework/theme/components/tree-grid/tree-grid-row.component.ts]: https://github.com/angular/components/pull/15440

File: src/framework/theme/components/layout/layout.component.spec.ts (hunk 9 above)
Comment (yggg) [reply src/framework/theme/components/layout/layout.component.spec.ts]: https://github.com/angular/components/pull/15440

File: src/framework/theme/components/cdk/table/row.ts (hunk 8 above)
Comment (yggg) [reply src/framework/theme/components/cdk/table/row.ts]: https://github.com/angular/components/pull/15440

File: src/framework/theme/components/cdk/table/row.ts (hunk 7 above)
Comment (yggg) [reply src/framework/theme/components/cdk/table/row.ts]: https://github.com/angular/components/pull/15440

File: src/framework/theme/components/cdk/table/row.ts (hunk 6 above)
Comment (yggg) [reply src/framework/theme/components/cdk/table/row.ts]: https://github.com/angular/components/pull/15440

File: scripts/gulp/tasks/build/compile/ts-configs.ts (hunk 5 above)
Comment (yggg) [reply scripts/gulp/tasks/build/compile/ts-configs.ts]: Yep

File: angular.json [hunk 11]
```diff
@@ -11,6 +11,7 @@
         "build": {
           "builder": "@angular-devkit/build-angular:browser",
           "options": {
+            "aot": true,
```
[the remark this answers is not in the document]
Comment (yggg) [reply angular.json]: According to the schema it's false by default
https://i.imgur.com/qN3qgMR.png

File: packages-smoke/angular.json (hunk 2 above)
Comment (yggg) [reply packages-smoke/angular.json]: According to the schema, it's false by default
https://i.imgur.com/qN3qgMR.png

**Comment by vishnu-dev** [comment]
Cool. Thanks again @yggg .

**Pull request merged**